lib/kafka/consumer.rb

```diff
@@ -352,6 +352,36 @@ def each_batch(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automaticall
     # @return [nil]
     def seek(topic, partition, offset)
       @offset_manager.seek_to(topic, partition, offset)
+      @fetcher.seek(topic, partition, offset)
+    end
+
+    # @param min_bytes [Integer] the minimum number of bytes to read before
+    #   returning messages from the server; if `max_wait_time` is reached, this
+    #   is ignored.
+    # @param max_bytes [Integer] the maximum number of bytes to read before
+    #   returning messages from each broker.
+    # @param max_wait_time [Integer, Float] the maximum duration of time to wait before
+    #   returning messages from the server, in seconds.
+    # @param max_retries [Integer] the maximum numbers of retries to fetch messages
+    #   before raising an error
+    # @return batch [Kafka::FetchedBatch] a message batch fetched from Kafka.
+    def poll(min_bytes: 1, max_bytes: 10485760, max_wait_time: 5, max_retries: 20)
+      @fetcher.configure(
+        min_bytes: min_bytes,
+        max_bytes: max_bytes,
+        max_wait_time: max_wait_time,
+      )
+
+      unless @running
+        @running = true
+        @fetcher.start
+      end
+
+      retry_or_raise(retries: max_retries) do
+        batches = fetch_batches
```

Comment:
Won't this risk dropping messages if `send_heartbeat_if_necessary` raises an exception? At the very least you'd need to stop and reset the fetcher, as it would keep pushing new fetch responses into the queue.